yarn-project/p2p/src/mem_pools/tx_pool/aztec_kv_tx_pool.ts

```diff
@@ -171,34 +193,23 @@ export class AztecKVTxPool extends (EventEmitter as new () => TypedEventEmitter<
             await this.#blockToDeletedMinedTxHash.deleteValue(originalBlock, key);
           }
         }
-
-        await this.#minedTxHashToBlock.set(key, blockHeader.globalVariables.blockNumber);
-
-        const tx = await this.getPendingTxByHash(hash);
-        if (tx) {
-          const nullifiers = tx.data.getNonEmptyNullifiers();
-          nullifiers.forEach(nullifier => minedNullifiers.add(nullifier.toString()));
-          minedFeePayers.add(tx.data.feePayer.toString());
-          pendingTxSize -= tx.getSize();
-          await this.removePendingTxIndices(tx, key);
-        }
       }
-      await this.#pendingTxSize.set(pendingTxSize);
+    });
 
-      await this.evictInvalidTxsAfterMining(txHashes, blockHeader, minedNullifiers, minedFeePayers);
+    await this.#store.transactionAsync(async () => {
+      try {
+        await this.#evictionManager.evictAfterNewBlock(blockHeader, uniqueMinedNullifiers, uniqueMinedFeePayers);
+      } catch (err) {
+        this.#log.warn('Unexpected error running evictAfterNewBlock', { err });
+      }
     });
-    // We update this after the transaction above. This ensures that the non-evictable transactions are not evicted
-    // until any that have been mined are marked as such.
-    // The non-evictable set is not considered when evicting transactions that are invalid after a block is mined.
-    this.#nonEvictableTxs.clear();
   }
 
-  public async markMinedAsPending(txHashes: TxHash[]): Promise<void> {
+  public async markMinedAsPending(txHashes: TxHash[], blockNumber: number): Promise<void> {
```

Comment:
Nit: let's rename the variable or add a comment explaining _what_ is this block number. Also, prefer using the `BlockNumber` branded type over a regular `number`.